### ISSUE TITLE
VS2017 RC - No signature help (method lists or parameter info )

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#1767]
On a fresh install of dev15 with F# tools installed, type the following in a new project:

```fsharp
open System

[<EntryPoint>]
let main argv = 
    let s = String.Format(
    
    0 // return an integer exit code
```

Note that there is no IntelliSense:

![no-intellisense-string-format](https://cloud.githubusercontent.com/assets/6309070/20402970/bcf6d812-acb3-11e6-98b3-ea6d518d5629.PNG)

Nor when the comma is pressed:

![no-intellisense-string-format-2](https://cloud.githubusercontent.com/assets/6309070/20402980/c7e28898-acb3-11e6-94b4-9b1197b7ec3b.PNG)

Note that this fails for more than just `String.Format`:

![no-method-overloads](https://cloud.githubusercontent.com/assets/6309070/20403012/f19dd8b8-acb3-11e6-8b65-36a1a8fbd8cb.PNG)

**Comment by dsyme** [comment]
@vladima Do you have pointers to which APIs to implement, or a skeleton implementation of this?  Thanks.  We have all the necessary information at hand, we just need to populate it

@vasily-kirichenko Would love to get this fixed today/tomorrow

**Comment by vladima** [comment]
to get signature help F# language service need to expose implementation of [ISignatureHelpProvider](http://source.roslyn.codeplex.com/#Microsoft.CodeAnalysis.Features/SignatureHelp/ISignatureHelpProvider.cs,c15027808b270a13), something like:
```F#
open Microsoft.CodeAnalysis.SignatureHelp

[<Shared>]
[<ExportSignatureHelpProvider("FSharpSignatureHelpProvider", FSharpCommonConstants.FSharpLanguageName)>]
type FSharpSignatureHelpProvider() =
    interface ISignatureHelpProvider with
        member this.IsTriggerCharacter(c) = c ='(' || c = '<' || c = ','
        member this.IsRetriggerCharacter(c) = c = ')' || c = '>'
        member this.GetItemsAsync(document, position, triggerInfo, cancellationToken) = // implement me
```

**Comment by dsyme** [comment]
Fixed by #1855
